### ISSUE TITLE
TealiumLogLevel.none not respected

### ISSUE THREAD

**Issue by tyler8691** [Tealium/tealium-swift#61]
When using tealium-swift 1.6.1, I am seeing that setting the following on `TealiumConfig` does not disable log output:

```
config.setLogLevel(logLevel: .none)
```

Looking deeper, it seems that `TealiumLogger` line 26 is comparing the `hashValue` of the `TealiumLogLevel` to determine if it should be logged. When debugging, `TealiumLogLevel.none.hashValue` is greater than `TealiumLogLevel.verbose.hashValue`, so logs are still being produced.

```
public func log(message: String, logLevel: TealiumLogLevel) -> String? {
        if logThreshold.hashValue >= logLevel.hashValue {
            var verbosity = ""
            if logLevel == .errors { verbosity = "ERROR: " }
            if logLevel == .warnings { verbosity = "WARNING: " }

            print("*** TEALIUM SWIFT \(TealiumValue.libraryVersion) *** Instance \(idString): \(verbosity)\(message)")
            return message
        }
        return nil
    }
```

Being as `TealiumLogLevel` already conforms to `Comparable`, removing the `hashValue` calls here seems to fix the issue. Setting `.none` results in no output, and setting `.verbose` results in verbose output.

**Comment by craigrouse** [comment]
Thanks for the feedback @tyler8691. We're looking into it and will come back to you ASAP.

**Comment by craigrouse** [comment]
@tyler8691 I can't reproduce this. TealiumLogLevel.none has a hashvalue of 0. Verbose is 3. If you initialize with a loglevel of none, the log threshold is always 0, which is never greater than the hashvalue of any of the other log levels. Can you provide any further information to help me debug?

I'm wondering if you have any code elsewhere in your project that may be conflicting with our implementation.

**Comment by craigrouse** [comment]
@tyler8691 ignore this message; I've just been able to reproduce, but the problem is intermittent. It seems hashValue is behaving inconsistently at different stages of the app lifecycle, which is most strange.

**Comment by tyler8691** [comment]
Yep that sounds like what we've seen, it only works intermittently. Seems like sometimes you'll be using the app for a while before it happens. Let me know if there's anything I can do to help!

**Comment by craigrouse** [comment]
Thanks for your help @tyler8691. I have created a fix. Should be released very soon.

**Comment by craigrouse** [comment]
@tyler8691 This has now been published in a new release, 1.6.3. Thank you for your patience and assistance with the fix. Please let us know if you have further issues.
